pkg/cluster/space/rook.go

```diff
@@ -0,0 +1,154 @@
+package clusterspace
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"code.cloudfoundry.org/bytefmt"
+	rookcli "github.com/rook/rook/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/replicatedhq/kurl/pkg/k8sutil"
+)
+
+const (
+	namespace = "rook-ceph"
+)
+
+// RookChecker checks if we have enough disk space to migrate volumes to rook.
+type RookChecker struct {
+	kcli  kubernetes.Interface
+	rcli  rookcli.Interface
+	cfg   *rest.Config
+	log   *log.Logger
+	srcSC string
+	dstSC string
+}
+
+// freeSpace attempts to get the ceph free space. returns the number of available bytes.
+func (r *RookChecker) freeSpace(ctx context.Context) (int64, error) {
+	pname, cname, err := r.getPoolAndClusterNames(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get ceph pool: %w", err)
+	}
+
+	pool, err := r.rcli.CephV1().CephBlockPools(namespace).Get(ctx, pname, metav1.GetOptions{})
+	if err != nil {
+		return 0, fmt.Errorf("failed to get pool %s: %w", pname, err)
+	}
+
+	// this should never happen but we better this than a division by zero.
+	if pool.Spec.Replicated.Size == 0 {
+		return 0, fmt.Errorf("pool replica size is zeroed")
+	}
+
+	cluster, err := r.rcli.CephV1().CephClusters(namespace).Get(ctx, cname, metav1.GetOptions{})
+	if err != nil {
+		return 0, fmt.Errorf("failed to get ceph cluster %s: %w", cname, err)
+	}
+
+	if cluster.Status.CephStatus == nil {
+		return 0, fmt.Errorf("failed to read ceph status (nil)")
+	}
+
+	availint64 := int64(cluster.Status.CephStatus.Capacity.AvailableBytes)
+	replicasint64 := int64(pool.Spec.Replicated.Size)
+	return availint64 / replicasint64, nil
+}
+
+// reservedSpace returns the total size of all volumes using the source storage class (srcSC).
+func (r *RookChecker) reservedSpace(ctx context.Context) (int64, error) {
```

Comment:
This feels like a function that shouldn't be in the rook checker specifically - should it go to k8sutil?

(there's probably one in openebs that does the same for a storageclass on a node too)

Reply:
There is a single function that is called by both: `k8sutil.PVSReservationPerNode()`. Rook simply sums the result while OpenEBS takes node names into account. Not sure if this change would bring us any improvements.